contracts/core/BaseAccount.sol

```diff
@@ -56,6 +59,7 @@ abstract contract BaseAccount is IAccount {
      *          (also hashes the entrypoint and chain-id)
      * @param aggregator the current aggregator. can be ignored by accounts that don't use aggregators
      * @return deadline the last block timestamp this operation is valid, or zero if it is valid indefinitely.
+     *      return SIG_VALIDATION_FAILED in case signature fails
```

Comment:
Well, I don't know about you, but I believe variables have names for a reason. And functions have return value for a reason. In this case I don't see a benefit in reusing the `deadline` variable as a `validation success` flag.

Solidity can totally return two values: a boolean success flag and an integer value.

Please do explain why did you choose to merge the values into one as I don't see it and this seems to me to make this trivial code confusing.

Reply:
1. solidity is expensive when handling return variables. 
2. a low number is "invalid". I just define a special low value as special invalid